tox.ini

```
@@ -1,30 +1,66 @@
+# Generated from:
+# https://github.com/zopefoundation/meta/tree/master/config/pure-python
 [tox]
-envlist = flake8,
-          py27,
-          py35,
-          py36,
-          py37,
-          py38,
-          pypy,
-          pypy3,
-          coverage
+envlist =
+    lint,
+    py27,
+    pypy,
+    py35,
+    py36,
+    py37,
+    py38,
+    pypy3,
+    coverage
 
 [testenv]
 usedevelop = true
+deps =
+    zope.testrunner
 commands =
     zope-testrunner --test-path=src []
 extras = test
 
-[testenv:coverage]
-basepython = python3.7
+[testenv:lint]
+basepython = python3
+skip_install = true
+deps =
+    flake8
+    check-manifest
+    check-python-versions
 commands =
-    coverage run -m zope.testrunner --test-path=src []
-    coverage report --fail-under=86
+    flake8 src setup.py
+    check-manifest
+    check-python-versions
+
+[testenv:coverage]
+basepython = python3
 deps =
     coverage
+    coverage-python-version
+    zope.testrunner
+setenv =
+    COVERAGE_PROCESS_START={toxinidir}/tox.ini
+commands =
+    coverage run -m zope.testrunner --test-path=src []
+    coverage html
+    coverage combine
```

You need to run `coverage combine` before `coverage html`.
```suggestion
    coverage combine
    coverage html
```

---

Awesome! At some time I have to read the manual carefully... same goes for the other 125 tools I use...

---

PR review is a perfectly cromulent way of knowledge transfer.

(It's why I comment on already merged PRs that are perfectly fine, but maybe show that the author didn't know of a particular nifty feature they could've used.)